src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -98,26 +101,33 @@ internal sealed class BuildUpToDateCheck : OnceInitializedOnceDisposed, IBuildUp
             _fileSystem = fileSystem;
         }
 
-        /// <summary>
-        /// Called on project load.
-        /// </summary>
-#pragma warning disable RS0030 // symbol ConfiguredProjectAutoLoad is banned
-        [ConfiguredProjectAutoLoad]
-#pragma warning restore RS0030 // symbol ConfiguredProjectAutoLoad is banned
-        [AppliesTo(ProjectCapability.DotNet + "+ !" + ProjectCapabilities.SharedAssetsProject)]
-        internal void Load()
+        public Task LoadAsync()
+        {
+            return InitializeAsync();
+        }
+
+        public Task UnloadAsync()
```

Comment:
What does it mean that this component doesn't unload itself? Why make it a `IProjectDynamicLoadComponent` if unloads do nothing?

Reply:
Agreed. This is also unrelated to the thing you are fixing. AutoLoad can be async, just have it return Task.

Reply:
Ah, I didn't know `[ConfiguredProjectAutoLoad]` would work with an async method. I'll back it out, although it probably does make sense to subscribe and unsubscribe from Dataflow in response to capabilities in future.

Reply:
@davkean are you sure `[ConfiguredProjectAutoLoad]` can return a task? It doesn't seem to work for me.

For now I'll make `UnloadAsync` clean up properly.

Reply:
MB, looks like only '[ProjectAutoLoad]' supports this not '[ConfiguredProjectAutoLoad]'

Reply:
Np. Take a look at the updated implementation and let me know if you see any issues.